LeaderboardBackend.Test/Features/Users/AccountConfirmationTests.cs

```diff
@@ -0,0 +1,148 @@
+
+        Client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await Client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+
+        res.Should().HaveStatusCode(HttpStatusCode.Conflict);
+    }
+
+    [Test]
+    public async Task ResendConfirmation_EmailFailedToSend()
+    {
+        Mock<IEmailSender> emailSenderMock = new();
+        emailSenderMock.Setup(e =>
+            e.EnqueueEmailAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())
+        ).Throws(new System.Exception());
+        HttpClient client = s_factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureTestServices(services =>
+            {
+                services.AddScoped<IEmailSender>(_ => emailSenderMock.Object);
+            });
+        })
+        .CreateClient();
+
+        IUserService userService = _scope.ServiceProvider.GetRequiredService<IUserService>();
+        CreateUserResult result = await userService.CreateUser(new()
+        {
+            Email = "email",
+            Password = "password",
+            Username = "username",
+        });
+        string token = _authService.GenerateJSONWebToken(result.AsT0);
+
+        client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+        res.Should().HaveStatusCode(HttpStatusCode.InternalServerError);
+    }
+
+    [Test]
+    public async Task ResendConfirmation_Success()
+    {
+        Mock<IEmailSender> emailSenderMock = new();
+        HttpClient client = s_factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureTestServices(services =>
+            {
+                services.AddScoped<IEmailSender>(_ => emailSenderMock.Object);
+            });
+        })
+        .CreateClient();
+        IUserService userService = _scope.ServiceProvider.GetRequiredService<IUserService>();
+        CreateUserResult result = await userService.CreateUser(new()
+        {
+            Email = "email",
+            Password = "password",
+            Username = "username",
+        });
+        string token = _authService.GenerateJSONWebToken(result.AsT0);
+
+        client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+        res.Should().HaveStatusCode(HttpStatusCode.OK);
+        emailSenderMock.Verify(x =>
+            x.EnqueueEmailAsync(
+                "email",
+                "Confirm Your Account",
+                It.IsAny<string>()
+            ),
+            Times.Once()
+        );
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        context.AccountConfirmations.FirstOrDefault(c => c.UserId == result.AsT0.Id)
+            .Should().NotBeNull();
```

Comment:
Check that the expiration date is 1 hour from now on the created entity.

`IClock` should be injected in `AccountConfirmationService`  instead of using a global instance, to allow `GetCurrentInstant()` to be mocked.

Reply:
Can't inject `IClock`. It won't build when I do so. Might need to use the global instance. However there's [NodaTime's testing lib](https://nodatime.org/2.1.x/userguide/testing) that we can use in tests instead.

Reply:
`IClock` is now injected, but current date is not mocked in the tests to verify the dates?
`CreatedAt` could be `1975/08/22 2PM` and `ExpirationDate` could be `1975/08/22 3 PM` for all we know.

Can use the library's fake implem for testing: https://nodatime.org/1.4.x/api/NodaTime.Testing.FakeClock.html